src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> insertNode;
+        if (size == 0) {
+            head = tail = new Node<>(null, value, null);
+            size++;
+            return;
+        }
+        insertNode = new Node<>(tail, value, null);
+        tail.next = insertNode;
+        tail = insertNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + "out of bounds, Size: " + size);
+        }
+        if (isEmpty() || index == size) {
+            add(value);
+            return;
+        }
+        Node<T> insertNode;
+        Node<T> node = getNode(index);
+        if (index == 0) {
+            insertNode = new Node<>(null, value, head);
+            head = insertNode;
+            head.prev = insertNode;
```

Comment:
why `head.prev = insertNode;`?

Reply:
because we add a node by index to the first position, so we need to move "head" to the right and link ex "head" prev to our new node.